lib/db/DB.ts

```diff
@@ -91,41 +68,63 @@ class DB {
     ]);
   }
 
-  getModels() {
-    return this.models;
-  }
-
-  close() {
+  public close = (): Bluebird<void> => {
     return this.sequelize.close();
   }
 
-  dropTables() {
+  public dropTables = (): Bluebird<void> => {
     return this.sequelize.drop();
   }
 
-  async truncate() {
+  public truncate = async (): Promise<void> => {
     await this.sequelize.transaction(async (t) => {
       const options = { raw: true, transaction: t };
-      await this.sequelize.query('SET FOREIGN_KEY_CHECKS = 0', null, options);
-      await this.sequelize.query('truncate table orders', null, options);
-      await this.sequelize.query('truncate table pairs', null, options);
-      await this.sequelize.query('truncate table currencies', null, options);
-      await this.sequelize.query('truncate table peers', null, options);
-      await this.sequelize.query('SET FOREIGN_KEY_CHECKS = 1', null, options);
+      await this.sequelize.query('SET FOREIGN_KEY_CHECKS = 0', options);
+      await this.sequelize.query('truncate table orders', options);
+      await this.sequelize.query('truncate table pairs', options);
+      await this.sequelize.query('truncate table currencies', options);
+      await this.sequelize.query('truncate table peers', options);
+      await this.sequelize.query('SET FOREIGN_KEY_CHECKS = 1', options);
     });
   }
 
-  async dbCreate() {
-    try {
-      this.sequelize = new Sequelize('', this.sequelize.options.username, this.sequelize.options.password, this.sequelize.options);
-      await this.sequelize.authenticate();
-      await this.sequelize.query(`CREATE DATABASE ${this.sequelize.options.database} CHARACTER SET utf8 COLLATE utf8_general_ci;`);
-      await this.sequelize.query(`USE ${this.sequelize.options.database}`);
-      this.close();
+  private createSequelizeInstance = (config: DBConfig|any): Sequelize.Sequelize => {
```

Comment:
`DBConfig|any` I believe is the same as just `any`. I think it's better to switch this to `DBConfig` and then make `database` nullable with `?` in the `DBConfig` declaration.

Reply:
I didn't want to do it because it can imply that `database` is an optional parameter for `xud`, wheres its an optional parameter only for this tiny helper method. The other option is to create another custom type, just for the "no-database" case. It will also decouple sequelize config from DBConfig, which is a good thing.

Reply:
I like your approach, thanks.